### PR TITLE
feat: sha input

### PR DESCRIPTION
**Pull request by m0n5t3r**
When the head SHA is different from the SHA that triggered the workflows (say, because there's a workflow that bumps the release and commits after the merge), the action will fail to find the upstream workflows; this adds a parameter called `sha` which can be set to `${{ github.event.workflow_run.head_commit.id }}`.

### CODE REVIEW

**Comment by m0n5t3r** [comment]
not sure what to do about the linter errors, some seem unrelated to my changes (markdown), one doesn't seem resolvable (the await thing), and the assignment error is because I got confused by the syntax and didn't want it to explode (this kind of javascript is not exactly my first language)

**Comment by ahmadnassri** [comment]
dont worry about the linter ... it complains a lot :P

File: action.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name: Workflow Run Wait
```
Comment (ahmadnassri) [inline action.yml:1]: need to revert these changes please

File: README.md [hunk 2]
```diff
@@ -84,6 +84,7 @@ jobs:
 | `github-token` | ❌        | `github.token` | The GitHub token used to call the GitHub API    |
 | `timeout`      | ❌        | `30000`        | timeout before we stop trying (in milliseconds) |
 | `delay`        | ❌        | `5000`         | delay between status checks (in milliseconds)   |
```
Comment (ahmadnassri) [inline README.md:86]: this needs to go in `docs/readme.md` plz 

(sorry this my templated appraoch to keep readme.md in sync)

File: action/lib/index.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ import workflows from './workflows.js'
...
-export default async function ({ token, delay, timeout }) {
+export default async function ({ token, delay, timeout, sha }) {
   let timer = 0
 
   // init octokit
   const octokit = github.getOctokit(token)
 
   // extract sha
-  const { sha, ref, runId: run_id } = github.context
+  const { ctx_sha, ref, runId: run_id } = github.context
```
Comment (ahmadnassri) [inline action/lib/index.js:21]: this variable is never used

**Review by ahmadnassri — CHANGES_REQUESTED** [review]

**Comment by ahmadnassri** [comment]
this is a good find ... thank you! left some reviews to address before I can merge

File: action.yml (hunk 1 above)
Comment (m0n5t3r) [reply action.yml:1]: yup, sorry, missed it

File: README.md (hunk 2 above)
Comment (m0n5t3r) [reply README.md]: done, hope it's ok

File: action/lib/index.js (hunk 3 above)
Comment (m0n5t3r) [reply action/lib/index.js]: I was afraid to remove it completely because the `runId: run_id` thing is something I haven't seen before :)

**Comment by m0n5t3r** [comment]
btw, I think the debug logging can be toned down a bit (https://github.com/ahmadnassri/action-workflow-run-wait/blob/master/action/lib/deduplicate.js#L18 tends to be very noisy and not very useful, resulting in logs being truncated randomly (?) for some reason) and the deduplication seems to not work, but I'll probably poke at it more next week

**Review by ahmadnassri — APPROVED** [review]
